app/services/supplementary-billing/process-billing-batch.service.js

```diff
@@ -10,19 +10,35 @@ const ProcessBillingPeriodService = require('./process-billing-period.service.js
 const UnflagUnbilledLicencesService = require('./unflag-unbilled-licences.service.js')
 
 async function go (billingBatch, billingPeriods) {
-  const currentBillingPeriod = billingPeriods[billingPeriods.length - 1]
   const { billingBatchId } = billingBatch
 
   try {
     // Mark the start time for later logging
     const startTime = process.hrtime.bigint()
+    let isBatchPopulated = false
+    let accumulatedLicenceIds = []
 
     await _updateStatus(billingBatchId, 'processing')
 
-    const chargeVersions = await _fetchChargeVersions(billingBatch, currentBillingPeriod)
-    const isPopulated = await ProcessBillingPeriodService.go(billingBatch, currentBillingPeriod, chargeVersions)
+    for (const billingPeriod of billingPeriods) {
+      const chargeVersions = await _fetchChargeVersions(billingBatch, billingPeriod)
+      const isPeriodPopulated = await ProcessBillingPeriodService.go(billingBatch, billingPeriod, chargeVersions)
 
-    await _finaliseBillingBatch(billingBatch, chargeVersions, isPopulated)
+      const licenceIdsForPeriod = _extractLicenceIds(chargeVersions)
+      const previousLicenceIds = accumulatedLicenceIds
+
+      accumulatedLicenceIds = previousLicenceIds.concat(licenceIdsForPeriod)
```

Comment:
We could do this in a couple of lines I think (but worth double-checking before committing!)
```suggestion
      const licenceIdsForPeriod = _extractLicenceIds(chargeVersions)
      accumulatedLicenceIds.push(...licenceIdsForPeriod)
```
Doing this would also let us declare `accumulatedLicenceIds` with `const`

Reply:
Cheers @StuAA78 I thought you'd have a better way of doing that :star: